gobblin-data-management/src/main/java/gobblin/data/management/retention/dataset/CleanableHiveDataset.java

```diff
@@ -0,0 +1,162 @@
+
+  public static final String SHOULD_DELETE_DATA_KEY = "gobblin.retention.hive.shouldDeleteData";
+  public static final String SHOULD_DELETE_DATA_DEFAULT = Boolean.toString(true);
+
+  public static final String VERSION_FINDER_CLASS_KEY = "version.finder.class";
+  public static final String DEFAULT_VERSION_FINDER_CLASS = DatePartitionHiveVersionFinder.class.getName();
+
+  public static final String SELECTION_POLICY_CLASS_KEY = "selection.policy.class";
+  public static final String DEFAULT_SELECTION_POLICY_CLASS = SelectBeforeTimeBasedPolicy.class.getName();
+
+  private final VersionSelectionPolicy hiveSelectionPolicy;
+  private final AbstractHiveDatasetVersionFinder hiveDatasetVersionFinder;
+
+  private final boolean simulate;
+  private final boolean shouldDeleteData;
+  private final FsCleanableHelper fsCleanableHelper;
+
+  public CleanableHiveDataset(FileSystem fs, HiveMetastoreClientPool clientPool, Table table, Properties jobProps, Config datasetConfig) throws IOException {
+    super(fs, clientPool, table, jobProps, datasetConfig);
+
+    try {
+      this.hiveSelectionPolicy =
+          (VersionSelectionPolicy) GobblinConstructorUtils.invokeFirstConstructor(
+              Class.forName(ConfigUtils.getString(datasetConfig, SELECTION_POLICY_CLASS_KEY, DEFAULT_SELECTION_POLICY_CLASS)),
+              ImmutableList.<Object> of(datasetConfig), ImmutableList.<Object> of(datasetConfig, jobProps), ImmutableList.<Object> of(jobProps));
+
+      this.hiveDatasetVersionFinder =
+          (AbstractHiveDatasetVersionFinder) GobblinConstructorUtils.invokeFirstConstructor(
+              Class.forName(ConfigUtils.getString(datasetConfig, VERSION_FINDER_CLASS_KEY, DEFAULT_VERSION_FINDER_CLASS)),
+              ImmutableList.<Object> of(this.fs, datasetConfig), ImmutableList.<Object> of(this.fs, jobProps));
+    } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException | InstantiationException | ClassNotFoundException e) {
+      throw new RuntimeException(e);
+    }
+
+    this.fsCleanableHelper = new FsCleanableHelper(fs, jobProps, datasetConfig, log);
+
+    this.shouldDeleteData = Boolean.valueOf(jobProps.getProperty(SHOULD_DELETE_DATA_KEY, SHOULD_DELETE_DATA_DEFAULT));
+    this.simulate = Boolean.valueOf(jobProps.getProperty(SIMULATE_KEY, SIMULATE_DEFAULT));
+  }
+
+  /**
+   * Drops the partitions selected by {@link #hiveSelectionPolicy}. Also deletes the data associated with it.
+   * <p>
+   * If an {@link Exception} occurs while processing a {@link Partition}, other {@link Partition}s will still be deleted.
+   * However, a {@link RuntimeException} is thrown at the end if there was at least one {@link Exception}.
+   * </p>
+   */
+  @Override
+  public void clean() throws IOException {
+
+    List versions = Lists.newArrayList(this.hiveDatasetVersionFinder.findDatasetVersions(this));
+
+    if (versions.isEmpty()) {
+      log.warn("No dataset version can be found. Ignoring.");
```

Comment:
Please include the name of the dataset in the log message (use `table.getCompleteName()` or a URI)